src/gennylib/src/encryption.cpp

```diff
@@ -357,6 +433,33 @@ void FLEEncryptedCollection::dropCollection(const mongocxx::client& client) cons
     client[_database][_collection].drop();
 }
 
+void QueryableEncryptedCollection::appendEncryptedFieldConfig(sub_document subdoc) const {
+    if (_fields.empty()) {
+        return;
+    }
+    subdoc.append(kvp("fields", [&](sub_array fieldsArray) {
+        for (const auto& [_, field] : _fields) {
+            fieldsArray.append([&](sub_document fieldDoc) { field.appendEncryptInfo(fieldDoc); });
+        }
+    }));
+}
+
+void QueryableEncryptedCollection::createCollection(const mongocxx::client& client) const {
+    auto createOpts = make_document(
+        kvp("encryptedFields", [&](sub_document subdoc) { appendEncryptedFieldConfig(subdoc); }));
+    BOOST_LOG_TRIVIAL(debug) << "Generated encryptedFieldConfig: "
+                             << bsoncxx::to_json(createOpts.view());
+    client[_database].create_collection(_collection, createOpts.view());
+}
+
+void QueryableEncryptedCollection::dropCollection(const mongocxx::client& client) const {
+    for (auto& suffix : {".esc", ".ecc", ".ecoc"}) {
```

Comment:
Shouldn't mongodb drop these automatically? It looks like it creates [them](https://www.mongodb.com/docs/upcoming/core/queryable-encryption/fundamentals/manage-collections/#metadata-collections)

Reply:
The current CXX driver does not have full support for QE, and automatic dropping of the auxilliary collections is one of the QE features that it lacks at the moment.

Reply:
Can you create a ticket to remove this code later and add it as a comment?

Reply:
done. see PERF-3306

Reply:
:+1: Thanks.